pyteal/ast/abi/address_test.py

```diff
@@ -0,0 +1,76 @@
+from ... import *
+
+options = CompileOptions(version=5)
+
+
+def test_AddressTypeSpec_str():
+    assert str(abi.AddressTypeSpec()) == "address"
+
+
+def test_AddressTypeSpec_is_dynamic():
+    assert (abi.AddressTypeSpec()).is_dynamic() is False
+
+
+def test_AddressTypeSpec_byte_length_static():
+    assert (abi.AddressTypeSpec()).byte_length_static() == abi.ADDRESS_LENGTH
+
+
+def test_AddressTypeSpec_new_instance():
+    assert isinstance(abi.AddressTypeSpec().new_instance(), abi.Address)
+
+
+def test_AddressTypeSpec_eq():
+    assert abi.AddressTypeSpec() == abi.AddressTypeSpec()
+
+    for otherType in (
+        abi.ByteTypeSpec,
+        abi.StaticArrayTypeSpec(abi.ByteTypeSpec(), 31),
```

Comment:
Hm, with 32 it fails the assert since they're seen as equal, I guess this means I should implement an __eq__ method on Address and raise an error if the other is not strictly an Address type?

Reply:
Yep exactly